service/security/shiro/src/main/java/org/eclipse/kapua/service/authentication/shiro/AuthenticationServiceShiroImpl.java

```diff
@@ -477,4 +489,34 @@ private String generateJwt(KapuaEid scopeId, KapuaEid userId, Date now, long ttl
         return jwt;
     }
 
+    private Boolean checkIfCredentialHasJustBeenLocked(AuthenticationToken authenticationToken) throws KapuaException {
+        String principal = (String) authenticationToken.getPrincipal();
+        User user = KapuaSecurityUtils.doPrivileged(() -> userService.findByName(principal));
+        Credential credential = null;
+        if (user != null) {
+            credential = KapuaSecurityUtils.doPrivileged(() -> {
+                CredentialListResult credentialList = credentialService.findByUserId(user.getScopeId(), user.getId());
+
+                if (credentialList != null && !credentialList.isEmpty()) {
+                    Credential credentialMatched = null;
+                    for (Credential c : credentialList.getItems()) {
+                        if (CredentialType.PASSWORD.equals(c.getCredentialType())) {
+                            credentialMatched = c;
+                            break;
+                        }
+                    }
+                    return credentialMatched;
+                } else {
+                    return null;
```

Comment:
Returning `null` here will produce a `NullPointerException` on the invocation on line 362.

Reply:
The `null` value in question is assigned to the `credential` variable in case that the `credentialList` is empty. Than that credential is checked and boolean value returned. 
I can't see how that would produce a `NullPointerException`. Am I missing something?

Reply:
Oh sorry, I missed the fact it was the return inside a lambda! So this is ok :)